### PR TITLE
Fix for misbehaving webviews

### PR DESCRIPTION
**Pull request by seanmtracey**
Webviews did not have initial states and were, in some circumstance, subject to race conditions that would result in webviews that were active not being visible and vice versa. Rather than adding/removing classes as the code deemed, the most recent state simply overwrites the lot.

Also adjusted the build pipeline to allow for deployment to point at both the test and live servers.

### CODE REVIEW

File: src/window.html [hunk 1]
```diff
@@ -10,8 +10,8 @@
 
 <body class="">
 
-	<webview minwidth="1024" minheight="768"></webview>
-	<webview minwidth="1024" minheight="768"></webview>
+	<webview minwidth="1024" minheight="768" partition="persist:ftlabs-screens"></webview>
```
Comment (JakeChampion) [inline src/window.html:13]: 👍

File: src/scripts/main.js [hunk 2]
```diff
@@ -165,8 +163,8 @@ window.onload = function() {
 
 		// Reset carousel countdown
 		window.removeEventListener('carousel-content-visible', onCarouselVisibleShowCountdown);
-		carouselCountdown.style.transform = 'scaleX(0)';
 		carouselCountdown.style.transition = 'none';
+		carouselCountdown.style.transform = 'scaleX(0)';
```
Comment (JakeChampion) [inline src/scripts/main.js:167]: Why this change?

File: src/scripts/main.js [hunk 3]
```diff
@@ -83,17 +83,15 @@ const views = [].slice.call(document.querySelectorAll('webview'));
...
 	window.dispatchEvent(visibleEvt);
 }
 
 function kickOutIframe(webView) {
-	webView.classList.remove('active');
-	webView.classList.remove('buffering');
-	webView.classList.add('done');
-	setTimeout(() => webView.src = 'about:blank', 500);
+	
+	webView.setAttribute('class', 'done');
```
Comment (JakeChampion) [inline src/scripts/main.js:93]: How come we are no longer removing the other classes?

File: src/scripts/main.js [hunk 4]
```diff
@@ -142,7 +140,7 @@ window.onload = function() {
 	const carouselCountdown = document.querySelector('#carousel-countdown');
 
 	let carousel;
-	const viewer = new Viewer('http://ftlabs-screens.herokuapp.com', chromeStorage);
+	const viewer = new Viewer('/* @echo host */', chromeStorage);
```
Comment (JakeChampion) [inline src/scripts/main.js]: Trailing whitespace

File: gulpfile.js [hunk 5]
```diff
@@ -1,15 +1,31 @@
...
-	});
+gulp.task('process', function() {
+	
+	var host = 'http://ftlabs-screens.herokuapp.com';
 
+	if(argv.deploy === 'test'){
+		host = 'http://ftlabs-screens-test.herokuapp.com';	
+	}
+	
+	gulp.src(['./src/scripts/main.js'])
```
Comment (JakeChampion) [inline gulpfile.js]: This cant be a string rather than an array of a string.

File: gulpfile.js [hunk 6]
```diff
@@ -1,15 +1,31 @@
...
+		.pipe( process( { context : {
+				host : host
+			}
+		}) )
+		.pipe( gulp.dest('./process/scripts/') )
+	;
+	
+	gulp.src(['./src/manifest.json'])
```
Comment (JakeChampion) [inline gulpfile.js]: Same as above.

File: gulpfile.js [hunk 7]
```diff
@@ -1,15 +1,31 @@
...
+		.pipe( process( { context : {
+				test : "-TEST"
```
Comment (JakeChampion) [inline gulpfile.js]: What is this doing?

File: src/scripts/main.js (hunk 2 above)
Comment (seanmtracey) [inline src/scripts/main.js:167]: I wanted that transform to occur after the transition had been removed, that way, there's no chance of the bar shooting from one end of the screen to another if a value somehow becomes assigned to it.

File: src/scripts/main.js (hunk 3 above)
Comment (seanmtracey) [inline src/scripts/main.js:93]: It didn't always take. By setting the class attribute, the previous values get removed and only the desired classes persist.

File: gulpfile.js (hunk 7 above)
Comment (seanmtracey) [inline gulpfile.js]: It was meant to add -TEST to the manifest.json file to indicate that the deployment points at the test servers, rather than the live ones. I couldn't get it to work, so I removed the placeholders for manifest.json, but I'll remove this now.

File: gulpfile.js (hunk 5 above)
Comment (seanmtracey) [inline gulpfile.js]: Yup, don't know why that's like that.

**Comment by JakeChampion** [comment]
LGTM 👍

File: src/scripts/main.js [hunk 8]
```diff
@@ -83,17 +83,15 @@ const views = [].slice.call(document.querySelectorAll('webview'));
...
+	this.setAttribute('class', 'active');
 	this.removeEventListener('load', webViewLoaded);
 	window.dispatchEvent(visibleEvt);
 }
 
 function kickOutIframe(webView) {
-	webView.classList.remove('active');
-	webView.classList.remove('buffering');
-	webView.classList.add('done');
-	setTimeout(() => webView.src = 'about:blank', 500);
```
Comment (AdaRoseCannon) [inline src/scripts/main.js:96]: You should still set it to about:blank somewhere lest the page sit in memory or carry on playing audio.

Comment (seanmtracey) [inline src/scripts/main.js:96]: Ah, that makes sense. I did wonder why this was being set. Will add now
